go/pkg/client/client.go

```diff
@@ -81,11 +81,11 @@ type Client struct {
 	// MaxBatchSize is maximum size in bytes of a batch request for batch operations.
 	MaxBatchSize MaxBatchSize
 	// DirMode is mode used to create directories.
-	DirMode os.FileMode
+	DirMode DirMode
 	// ExecutableMode is mode used to create executable files.
-	ExecutableMode os.FileMode
+	ExecutableMode ExecutableMode
 	// RegularMode is mode used to create non-executable files.
-	RegularMode os.FileMode
+	RegularMode RegularMode
```

Comment:
@ola-rozenfeld 
These changes will affect the existing usages. are you ok with changing them?

```
❯ go build ./...
# github.com/bazelbuild/remote-apis-sdks/go/pkg/client
go/pkg/client/bytestream.go:93:68: cannot use c.RegularMode (type RegularMode) as type os.FileMode in argument to os.OpenFile
go/pkg/client/cas.go:792:68: cannot use c.RegularMode (type RegularMode) as type os.FileMode in argument to os.OpenFile
go/pkg/client/cas.go:1117:33: cannot use c.DirMode (type DirMode) as type os.FileMode in argument to os.MkdirAll
go/pkg/client/cas.go:1122:46: cannot use c.DirMode (type DirMode) as type os.FileMode in argument to os.MkdirAll
go/pkg/client/cas.go:1153:9: cannot use c.ExecutableMode (type ExecutableMode) as type RegularMode in assignment
go/pkg/client/cas.go:1159:55: cannot use perm (type RegularMode) as type os.FileMode in argument to copyFile
go/pkg/client/cas.go:1256:10: cannot use c.ExecutableMode (type ExecutableMode) as type RegularMode in assignment
go/pkg/client/cas.go:1258:44: cannot use perm (type RegularMode) as type os.FileMode in argument to ioutil.WriteFile
go/pkg/client/cas.go:1280:29: cannot use c.ExecutableMode (type ExecutableMode) as type os.FileMode in argument to os.Chmod
go/pkg/client/cas.go:1287:9: cannot use c.ExecutableMode (type ExecutableMode) as type RegularMode in assignment
```

Reply:
Yeah, a simple cast should do (like we do e.g. https://github.com/bazelbuild/remote-apis-sdks/blob/master/go/pkg/client/cas.go#L1310)
Thank you!

Reply:
os.FileMode(...) everywhere didn't look good to me.
I cast os.FileMode(...) in Apply() instead.